source/components/JsxParser.js

```diff
@@ -120,6 +120,15 @@ export default class JsxParser extends Component {
           case '!': return !expression.argument.value
         }
         return undefined
+      case 'ArrowFunctionExpression':
+        if (expression.async || expression.generator) {
+          this.props.onError(new Error('Async and generator arrow functions are not supported.'))
+        }
+        const params = expression.params.map(this.parseExpression)
+        const body = // Not quite sure how to parse the body... it would need contextual values of local vars
+        // Not sure if we can use new Function since it only takes strings
+        // eslint-disable-next-line no-new-func
+        return new Function(...params, body)
```

Comment:
So, this is definitely where the complexity lies. We need to handle a few things, here - which is where I stopped, as well. https://github.com/acornjs/acorn/blob/master/acorn/src/expression.js is a reference, btw, to how Acorn parses functions of both kinds (arrow and function).

We should support:
```
<JsxParser
  bindings={{ foo: 'bar' }}
  jsx={`<div onClick={event => console.log(event.target, foo)} />`}
/>
```

And `onClick`, this should, if correct, log out a reference to the DOM node of the `div`, as well as the string `'bar'`.

This means that when composing the function, we need to evaluate each of the expressions used within it against both `this.props.bindings` _and_ against the param names of the function itself.

Reply:
When will support this feathure?

Reply:
@rxwater - the PR has been open for a long time, primarily because there has been no activity to support it, and use cases which would require this are generally achievable with less insecure solutions. So, unfortunately, there is no plan to move forward with it at this time.

I've left it open since it was started, with the help wanted tag, in case someone wants to take it on and address the various security issues involved.